bin/pytorch_inference/Main.cc

```diff
@@ -78,7 +79,8 @@ void writeTensor(torch::TensorAccessor<float, 1> accessor,
     jsonWriter.EndArray();
 }
 
-void writeTensor(torch::TensorAccessor<float, 2> accessor,
+template<typename T>
+void writeTensor(torch::TensorAccessor<T, 2>& accessor,
```

Comment:
Unrelated to this change, but it feels slightly cleaner to have this implemented in terms of `writeTensor(torch::TensorAccessor<T, 1>& accessor,...`. `operator[]` on `torch::TensorAccessor<T, N>` returns ``torch::TensorAccessor<T, N-1>` so this should just work. (This could be made fully generic, but not sure it is worthwhile at the moment.)

Reply:
This is much better. In the nested for loop I was using a new accessor `accessor[i]` was created for every access `accessor[i][j]`, that object is now reused.